File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

to_rna(Strand) -> [convert(X) || X <- Strand].

convert($G) -> $C;
convert($C) -> $G;
convert($T) -> $A;
convert($A) -> $U;
convert(_) -> throw(bad_nucleotide).

File: prolog/rna-transcription/rna_transcription.pl
rna_transcription(Rna, Dna) :-
    string_chars(Rna, Chars),
    maplist(convert, Chars, DnaChars),
    string_chars(Dna, DnaChars).

convert('G', 'C').
convert('C', 'G').
convert('T', 'A').
convert('A', 'U').
